ENG-12821: Updates to describe ipywidgets (#532)
opendatahub-io/opendatahub-documentation: modules/managing-ray-clusters-from-within-a-jupyter-notebook.adoc

** Open the Ray dashboard directly from the notebook, to view the submitted jobs.

+
You can add these controls to existing notebooks, or manage the Ray clusters from a separate notebook.

The `3_widget_example.ipynb` demo notebook shows all of the available interactive browser controls.
In the example in this procedure, you create a new notebook to manage the Ray clusters, similar to the example provided in the `3_widget_example.ipynb` demo notebook.

.Prerequisites
ifndef::upstream[]
* You can access a data science cluster that is configured to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/managing_openshift_ai/managing_distributed_workloads[Managing distributed workloads].
endif::[]
ifdef::upstream[]
* You can access a data science cluster that is configured to run distributed workloads as described in link:{odhdocshome}/managing-odh/#_managing_distributed_workloads[Managing distributed workloads].
endif::[]

* You can access the following software from your data science cluster:
** A Ray cluster image that is compatible with your hardware architecture
** The data sets and models to be used by the workload

> ** The Python dependencies for the workload, either in a Ray image or in your own Python Package Index (PyPI) server

Author: These lines are for a list that is at a different level than the list items in lines 17 and 18. The items in this second-level list are all fragments, so no period needed, according to _IBM Style_.

endif::[]

.Procedure

ifndef::upstream[]
. Run all of the demo notebooks in the order indicated by the file-name prefix (`0_`, `1_`, and so on), as described in link:{rhoaidocshome}{default-format-url}/working_with_distributed_workloads/running-distributed-workloads_distributed-workloads#running-the-demo-notebooks-from-the-codeflare-sdk_distributed-workloads[Running the demo notebooks from the CodeFlare SDK].
endif::[]
ifdef::upstream[]
. Run all of the demo notebooks in the order indicated by the file-name prefix (`0_`, `1_`, and so on), as described in link:{odhdocshome}/working-with-distributed-workloads/#running-the-demo-notebooks-from-the-codeflare-sdk_distributed-workloads[Running the demo notebooks from the CodeFlare SDK].
endif::[]

. In each demo notebook, when you run the cluster configuration step, the following interactive controls are automatically shown in the notebook:

* *Cluster Up*: You can click this button to start the Ray cluster. 
This button is equivalent to the `cluster.up()` command. 

> A message is displayed to indicate whether the cluster was successfully created or not.

Reviewer: ```suggestion
A message is displayed that indicates whether the cluster was successfully created.
```
Minor nitpick - "whether" implies "or not"

* *Cluster Down*: You can click this button to delete the Ray cluster. 
This button is equivalent to the `cluster.down()` command.

> The cluster is deleted immediately, you are not prompted to confirm the deletion.

Reviewer: ```suggestion
The cluster is deleted immediately. You are not prompted to confirm the deletion.
```
or
```suggestion
The cluster is deleted immediately; you are not prompted to confirm the deletion.
```

> A message is displayed to indicate whether the cluster was successfully deleted or not.

Reviewer: ```suggestion
A message is displayed that indicates whether the cluster was successfully deleted.
```

+
If you have access to another project, you can list the Ray clusters in that project by specifying the project name as shown in the following example:
+
.Code to view clusters in another project
[source,bash]
----
view_clusters("my_second_project")
----

.. Click *File > Save Notebook As*, enter `demo-notebooks/guided-demos/manage_ray_clusters.ipynb`, and click *Save*.

. In the `demo-notebooks/guided-demos/manage_ray_clusters.ipynb` notebook, select each cell in turn, and click *Run > Run selected cell*.

. When you run the cell with the `view_clusters()` function, the notebook shows the following information about the existing Ray clusters:

* *Select an existing cluster* 
+
Under this heading, a toggle button is shown for each existing cluster. 
Click a cluster name to select the cluster.
The cluster details section is updated to show details about the selected cluster; for example, cluster name, {productname-short} project name, cluster resource information, and cluster status.

* *Delete cluster*
+
Click this button to delete the selected cluster.
This button is equivalent to the *Cluster Down* button.

> The cluster is deleted immediately, you are not prompted to confirm the deletion.

Reviewer: Same feedback as previous

> A message is shown to confirm the deletion, and the corresponding button is no longer shown under the *Select an existing cluster* heading.

Reviewer: > A message is shown to confirm the deletion,

Doesn't this contradict the previous sentence?
